java/rocksjni/options.cc

```diff
@@ -232,7 +232,7 @@ void Java_org_rocksdb_Options_setStatistics(
     JNIEnv* env, jobject jobj, jlong jhandle, jlong jstatistics_handle) {
   auto* opt = reinterpret_cast<rocksdb::Options*>(jhandle);
   auto* pSptr =
-      reinterpret_cast<std::shared_ptr<rocksdb::StatisticsJni>*>(
+      reinterpret_cast<std::shared_ptr<rocksdb::Statistics>*>(
```

Comment:
I guess my question, is more about whether that is a safe thing to do in C++, i.e. will the cast create an object with the same memory semantics as previously.